aspnetcore/hosting/ihostingstartup.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Add app features from an external assembly using IHostingStartup in ASP.NET Core
+author: guardrex
+description: Discover how to add features to an ASP.NET Core app from an external assembly using an IHostingStartup implementation.
+ms.author: riande
+manager: wpickett
+ms.date: 12/07/2017
+ms.topic: article
+ms.custom: mvc
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: hosting/ihostingstartup
+---
+# Add app features from an external assembly using IHostingStartup in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+An [IHostingStartup](/dotnet/api/microsoft.aspnetcore.hosting.ihostingstartup) implementation allows adding features to an app at startup from outside of the app's `Startup` class. For example, an external tooling library can use an `IHostingStartup` implementation to provide additional configuration providers or services to an app. `IHostingStartup` *is available in ASP.NET Core 2.0 and later.*
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/hosting/ihostingstartup/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## Implement IHostingStartup features
+
+### Create the assembly
+
+An `IHostingStartup` feature is deployed as an assembly based on a console app without an entry point. The assembly references the [Microsoft.AspNetCore.Hosting.Abstractions](https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.Abstractions/) package:
+
+[!code-xml[Main](ihostingstartup/snapshot_sample/StartupFeature.csproj)]
+
+A [HostingStartupAttribute](/dotnet/api/microsoft.aspnetcore.hosting.hostingstartupattribute) identifies a class as an implementation of `IHostingStartup` for loading and execution when building the [IWebHost](/dotnet/api/microsoft.aspnetcore.hosting.iwebhost). In the following example, the namespace is `StartupFeature`, and the class is `StartupFeatureHostingStartup`:
```

Comment:
`A [HostingStartup](/dotnet/api/microsoft.aspnetcore.hosting.hostingstartupattribute) attribute identifies a class as an implementation `

A [HostingStartup](/dotnet/api/microsoft.aspnetcore.hosting.hostingstartupattribute) attribute identifies a class as an implementation

We usually write attributes that way give that's how they're used in code.